### PR TITLE
EVG-2019 Set last command type before timeout task runs

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: proto/agent.go [hunk 1]
```diff
@@ -232,17 +237,20 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 	return resp, nil
 }
 
-func (a *Agent) endTaskResponse(tc *taskContext, status string, timeout bool) *apimodels.TaskEndDetail {
-	detail := &apimodels.TaskEndDetail{
+func (a *Agent) getCurrentCommandType(tc *taskContext) string {
+	if tc.taskConfig != nil {
+		return tc.currentCommand.GetType(tc.taskConfig.Project)
+	}
+	return ""
```
Comment (john-m-liu) [inline proto/agent.go]: Should this line return the default instead?

File: proto/agent_test.go [hunk 2]
```diff
@@ -212,7 +212,6 @@ func (s *AgentTestSuite) TestAbort() {
 	err := s.a.runTask(context.Background(), s.tc)
 	s.NoError(err)
 	s.Equal(evergreen.TaskFailed, s.mockCommunicator.EndTaskResult.Detail.Status)
-	s.Equal("initial task setup", s.mockCommunicator.EndTaskResult.Detail.Description)
```
Comment (john-m-liu) [inline proto/agent_test.go]: Did this assert need to be removed?

File: proto/agent.go (hunk 1 above)
Comment (bsamek) [reply proto/agent.go]: done

File: proto/agent_test.go (hunk 2 above)
Comment (bsamek) [reply proto/agent_test.go]: done

**Pull request merged**